waku/README.md

```diff
@@ -0,0 +1,301 @@
+
+Light nodes MUST NOT forward any incoming messages, they MUST only send their own messages. Light nodes SHOULD not connect to other light nodes, this would result in messages being dropped between the two.
+
+Light nodes are identified using the handshake.
+
+<!-- TODO: Add details on handshake -- >
+
+**Security consideration**:
+Running a node as a light node mode impacts privacy due to the fact that it becomes identifiable what nodes care about if they aren't relaying traffic.
+
+<!-- TODO: Consider moving all security considerations to end section of document -->
+
+### Mailserver and client
+
+<!-- TODO: Clean up vocabulary, is it mailserver and mailserver client? mailclient? mailserver node? historynode? etc -->
+
+## Rationale
+
+Packet codes 0x00 and 0x01 are already used in all Whisper versions.
+
+Packet code 0x02 will be necessary for the future development of Whisper. It will provide possiblitity to adjust the PoW requirement in real time. It is better to allow the network to govern itself, rather than hardcode any specific value for minimal PoW requirement.
+
+Packet code 0x03 will be necessary for scalability of the network. In case of too much traffic, the nodes will be able to request and receive only the messages they are interested in.
+
+Packet codes 0x7E and 0x7F may be used to implement Whisper Mail Server and Client. Without P2P messages it would be impossible to deliver the old messages, since they will be recognized as expired, and the peer will be disconnected for violating the Whisper protocol. They might be useful for other purposes when it is not possible to spend time on PoW, e.g. if a stock exchange will want to provide live feed about the latest trades.
+
+## Backwards Compatibility
+
+This EIP is compatible with Whisper version 6. Any client which does not implement certain packet codes should gracefully ignore the packets with those codes. This will ensure the forward compatibility. 
+
+### WakuWhisper bridging
+
+`waku/0` and `shh/6` are different DevP2P subprotocols. In order to achieve backwards compatibility, bridging is required. It works as follows.
+
+**Roles:**
+- Waku client A, only Waku capability
+- Whisper client B, only Whisper capability
+- WakuWhisper bridge C, both Waku and Whisper capability
+
+**Flow:**
+1. A posts message; B posts message.
+2. C picks up message from A and B and relays them both to Waku and Whisper.
+3. A receives message on Waku; B on Whisper.
+
+**Note**: This flow means if another bridge C1 is active, we might get duplicate relaying for a message. I.e. Whisper(<>Waku<>Whisper)<>Waku, A-C1-C2-B. Theoretically this bridging chain can get as long as TTL permits.
+
+## Forwards Compatibility
+
+It is desirable to maintain forward compatibility between `waku/0` and future version of waku. Here we outline some concerns and strategy for this.
+
+<!-- TODO: Think about how to maintain forwards capability for waku/v0 -> v1 -> v2, etc. -->
```

Comment:
I don't see it necessary for `waku/v0`. I mean we can have a goal to do that but if it turns out in e.g. `v2` that we want to drop something from `v0` that should be possible.

Also, I don't see any value in supporting all the versions down. Like why `v5` would need to be backward compatible with `v1`. It may opinion a healthy evolvement is more important that staying backward compatible with a few years old versions. There should be a line drawn.

Reply:
We need to be compatible with Status v1.

Compatibility with future version: that's where DEPRECATION comes in. It doesn't mean we won't keep compatibility with v0 forever. I agree deprecating things once it makes sense is healthy.

Reply:
I wrote this is as 'have a strategy for' to leave the door open for deprecation in the future.

Also added TODO for deprecation notice and recommendation for clients

If we have ideas for specific stuff we ewant to drop support for in future it'd be good to note these in https://github.com/vacp2p/specs/issues/41